src/BBT.Maybe/MaybeStruct.cs

```diff
@@ -59,7 +60,7 @@ private MaybeStruct(SerializationInfo info, StreamingContext context)
         /// <param name="b">Maybe to compare.</param>
         public static bool operator !=(MaybeStruct<T> a, MaybeStruct<T> b)
         {
-            return a != b;
+            return !(a == b);
```

Comment:
Is this change intended?

Reply:
yes, recursive call of operator causes stack overflow exception.